packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -192,6 +193,10 @@ Future<void> buildMacOS({
     }
   }
 
+  if (!codesign) {
+    globals.printStatus('Codesigning disabled with --no-codesign.');
+  }
```

Comment:
I would move this into the command and make the message more like the iOS version:

https://github.com/flutter/flutter/blob/61fe9b6b67c08b03f6e97b015ba4d18dbd072a41/packages/flutter_tools/lib/src/commands/build_ios.dart#L741-L746